spec/services/publish/courses/assign_tda_attributes_service_spec.rb

```diff
@@ -3,19 +3,54 @@
 require 'rails_helper'
 
 RSpec.describe Publish::Courses::AssignTdaAttributesService do
-  let(:course) { create(:course, study_mode: 'part_time', funding_type: 'fee', can_sponsor_skilled_worker_visa: true, can_sponsor_student_visa: true) }
+  subject(:service) { described_class.new(course) }
+
+  let(:course) do
+    create(
+      :course,
+      study_mode: 'part_time',
+      funding_type: 'fee',
+      can_sponsor_skilled_worker_visa: true,
+      can_sponsor_student_visa: true
+    )
+  end
 
   describe '#call' do
-    it 'updates the course attributes and returns true' do
-      expect(described_class.new(course).call).to be_truthy
-      expect(course.study_mode).to eq('full_time')
-      expect(course.funding_type).to eq('apprenticeship')
-      expect(course.can_sponsor_skilled_worker_visa).to be(false)
-      expect(course.can_sponsor_student_visa).to be(false)
-      expect(course.additional_degree_subject_requirements).to be(false)
-      expect(course.degree_subject_requirements).to be_nil
-      expect(course.degree_grade).to eq('not_required')
-      expect(course.degree_type).to eq('undergraduate')
+    context 'when course does not have enrichments' do
```

Comment:
Since we've specified postgrad below, should we specify the `degree_type` here?

Reply:
Done